plugins/kubernetes-backend/src/kubernetes-auth-translator/AzureIdentityKubernetesAuthTranslator.ts

```diff
@@ -31,11 +42,23 @@ export class AzureIdentityKubernetesAuthTranslator
       clusterDetails,
     );
 
-    const credentials = new DefaultAzureCredential();
+    if (!this.accessToken || this.tokenExpired()) {
```

Comment:
Token presence check is also done in `tokenExpired`, perhaps it could be refactored to be `this.needsTokenRefresh()`?

Reply:
I did it because this.accessToken is used a few lines below and TypeScript complaints that it's possibly null, I can `this.accessToken!` but I don't like it :) Any other suggestion?